src/services/geometry/richgeo/RichGeo_service.h

```diff
@@ -30,6 +28,9 @@ class RichGeo_service : public JService {
     RichGeo_service(JApplication *app) : m_app(app) {}
     virtual ~RichGeo_service();
 
+    // return pointer to the main DD4hep Detector
+    dd4hep::Detector *GetDD4hepGeo() { return m_dd4hepGeo; };
```

Comment:
@veprbl should `GetDD4hepGeo` be virtual too?

Reply:
Yes